js/src/ui/Actionbar/Export/export.js

```diff
@@ -38,19 +39,45 @@ class ActionbarExport extends Component {
         className={ className }
         icon={ <FileDownloadIcon /> }
         label='export'
-        onClick={ this.onDownloadBackup } />
+        onClick={ this.handleExport }
+      />
     );
   }
 
-  onDownloadBackup = () => {
+  onDownloadBackup = (filetype) => {
     const { filename, content } = this.props;
 
-    const text = (typeof content === 'string')
-      ? content
-      : JSON.stringify(content, null, 4);
+    const text = this.contentAsString(content, filetype);
+    const extension = this.getExtension(filetype);
 
     const blob = new Blob([ text ], { type: 'text/plain;charset=utf-8' });
-    FileSaver.saveAs(blob, filename);
+    FileSaver.saveAs(blob, `${filename}.${extension}`);
+  }
+
+  getExtension = (filetype) => {
+    switch (filetype) {
+      case 'json':
+        return filetype;
+      default:
+        return 'txt';
+    }
+  }
+
+  contentAsString = (data, filetype) => {
+    if (typeof data === 'string') {
+      return data;
+    }
+
+    switch (filetype) {
+      case 'json':
+        return JSON.stringify(data, null, 4);
+      default:
+        return data.toString();
+    }
+  }
+
+  handleExport = () => {
```

Comment:
is this abstraction worth it? also `getExtension` feels like we're not going to need it for new.